src/Features/CSharp/Portable/InlineParameterNameHints/CSharpInlineParameterNameHintsService.cs

```diff
@@ -76,6 +76,16 @@ public CSharpInlineParameterNameHintsService()
         /// <returns>true when the adornment should be added</returns>
         private static bool IsExpressionWithNoName(ExpressionSyntax arg)
         {
+            if (arg.Parent.Parent is BracketedArgumentListSyntax bracketList)
+            {
+                // Do not need to add hints for array accesses if the length is 1
+                if (bracketList.Arguments.Count == 1)
+                {
+                    return false;
+                }
+
+                return true;
```

Comment:
This is just a suggestion.

```suggestion
                return bracketList.Arguments.Count > 1;
```